### ISSUE TITLE
"No bib files found!" pop-up on `\cite{}`

### ISSUE THREAD

**Issue by kortschak** [SublimeText/LaTeXTools#1084]
Recently (past week or so) LaTeXTools has been popping up the "No bib files found!" error dialog.

Installed version of LaTexTools is v3.13.5. SublimeText 3 Build 3126.

Console:
```
TEX root: '/path/to/document.tex'
Bib files found: 
[]
error: No bib files found!
```

Shell:
```
$ ls /path/to | head -n 1
bibliography.bib
```

**Comment by r-stein** [comment]
How do you add the bibliography? Does this also happen after you clear the cache (C-l, C-d, C-c)?

**Comment by kortschak** [comment]
>How do you add the bibliography?

The bibliography is included with `\bibliography{bibliography}` and the bibliography.bib file is placed in the working directory. This worked until recently (it would be surprising to me for that to not work).

>Does this also happen after you clear the cache (C-l, C-d, C-c)?

No change.

**Comment by r-stein** [comment]
Have you wrapped `\bibliography{bibliography}` inside an other command? Can you please provide a minimal working example, which produces the error?

**Comment by kortschak** [comment]
My minimal reproducer is

```
\begin{document}
\cite{}
\verb|\end{document}|
\bibliography{bibliography}
\end{document}
```

with a valid `bibliography.bib`.

Removing the `\verb|\end{document}|` fixes the problem, so I suspect that the quoted `\end{document}` is being parsed as `\end{document}` rather than text.

**Comment by r-stein** [comment]
Yes you are correct the analysis stops at `\end{document}` and is not smart enough to detect such cases.
You can try to remove [that line](https://github.com/SublimeText/LaTeXTools/blob/v3.13.5/latextools_utils/analysis.py#L431). Not sure what would be the best to solve this in general.

**Comment by kortschak** [comment]
That text is only there in placeholder text in my case, so this is not a huge issue. For others, a work around is to replace that with `\verb|\|\verb|end{document}|`. Ugly, but workable.

**Comment by r-stein** [comment]
> Removing that whole condition also fixes the problem. It would seem that erroring in greediness would be better that in the other direction.

I tend to agree on that.

**Comment by lrdegeest** [comment]
Similar problem. No .tex files can detect the bibliography one month after it was worked fine.

**Comment by lrdegeest** [comment]
Exhibit A: 
![screen shot 2017-04-27 at 10 54 02 pm](https://cloud.githubusercontent.com/assets/10857097/25512575/e53c7a8a-2b9c-11e7-806e-c9562a664110.png)

**Comment by r-stein** [comment]
@lrdegeest Do you also have `\end{document}` inside a verb inside your document? Otherwise it would also be good, if you good provide a minimal working example.

**Comment by lrdegeest** [comment]
Negative. MWE:

```
\begin{document}
Consider the lobster \citep{}
\bibliography{bibfile}
\end{document}
```

**Comment by r-stein** [comment]
@lrdegeest Then it is an other bug and it would be nice if you could create a new issue as I can't reproduce it.

**Comment by lrdegeest** [comment]
Sure, but one last point. The problem is also sensitive to the _name_ of a file. "test.tex" works; "experiment.tex" or "asasdfl.tex" (random letters) do not.

**Comment by r-stein** [comment]
Can you press C-l, C-d, C-c and then check whether it still doesn't work?

**Comment by lrdegeest** [comment]
Already tried clearing the cache. No effect.

**Comment by r-stein** [comment]
Do you have any errors in the console ``ctrl+` ``?

**Comment by lrdegeest** [comment]
Yes:

```
Running "'' -version"
Traceback (most recent call last):
  File "/Users/LawrenceDeGeest/Library/Application Support/Sublime Text 3/Packages/LaTeXTools/st_preview/preview_utils.py", line 99, in _update_gs_version
    raw_version = check_output([_GS_COMMAND, '-version'])
  File "/Users/LawrenceDeGeest/Library/Application Support/Sublime Text 3/Packages/LaTeXTools/latextools_utils/external_command.py", line 348, in check_output
    show_window=show_window
  File "/Users/LawrenceDeGeest/Library/Application Support/Sublime Text 3/Packages/LaTeXTools/latextools_utils/external_command.py", line 268, in execute_command
    show_window=show_window
  File "/Users/LawrenceDeGeest/Library/Application Support/Sublime Text 3/Packages/LaTeXTools/latextools_utils/external_command.py", line 221, in external_command
    cwd=cwd
  File "./python3.3/subprocess.py", line 819, in __init__
  File "./python3.3/subprocess.py", line 1370, in _execute_child
  File "./python3.3/os.py", line 791, in fsencode
TypeError: expect bytes or str, not NoneType
```

**Comment by lrdegeest** [comment]
However, if I rename the file to "test.tex", changing nothing else, it works.

**Comment by lrdegeest** [comment]
In fact, I am noticing that the problem arises in any tex file if if move it. This should not be a problem because I provide the absolute path to the bibliography, and anyway, it worked perfectly fine before.

**Comment by r-stein** [comment]
That error is not related to that (and not really problematic).

We have changed things in that completion, because we now use a backward analysis to extract information of the document instead rewritting that feature everywhere (this adds support for the import package). It doesn't seem to support absolute paths.

Can you try to change [this line](https://github.com/SublimeText/LaTeXTools/blob/v3.13.6/latex_cite_completions.py#L259) to:

``` py
            if not os.path.isabs(res):
                candidate_file = os.path.normpath(os.path.join(rootdir, res))
            else:
                candidate_file = res
```

You can use [PackageResourceViewer](https://packagecontrol.io/packages/PackageResourceViewer) to edit the files.

**Comment by lrdegeest** [comment]
On the contrary, it is quite problematic! 

Changing the line in the python file has no effect. For example, I create an empty file called "results.tex" and add a path to the bibliography. Nada:

![screen shot 2017-04-28 at 2 10 15 pm](https://cloud.githubusercontent.com/assets/10857097/25541466/bf006b44-2c1c-11e7-8b94-e9bfc8be3235.png)

but if I change the file name to "test.tex" (`mv results.tex test.tex`), suddenly it works:

![screen shot 2017-04-28 at 2 10 42 pm](https://cloud.githubusercontent.com/assets/10857097/25541508/e885509c-2c1c-11e7-9e0e-aa2cc9f660d4.png)

**Comment by lrdegeest** [comment]
Just to be clear, pdflatex and bibtex still compile a bibliography despite the message "No bib file found."

**Comment by ig0774** [comment]
@r-stein wasn't saying the "No bib files" message is problematic. It's the output from the console you showed that indicates a problem finding Ghostscript on your machine, which is completely unrelated to the bibliography problem.

Unfortunately, it's not possible to reproduce the error you're showing from the MWE you provided on my machine, which just indicates that the problem isn't obvious just from the example you provided. Can you be a bit more detailed as to the steps you're taking to produce this issue? You seem to be renaming the file via `mv`; is this change immediately reflected in ST or do you somehow reopen the file once it's been renamed? Being specific and clear will help us to help you.

**Comment by ig0774** [comment]
Sorry, first line should be *isn't* instead of *is*.

**Comment by lrdegeest** [comment]
Thanks a lot for your help. I recreated the problem with a fresh example. I created a directory called "mypaper":

```
└── mypaper
    ├── conclusion.tex
    ├── intro.tex
    ├── main.tex
    └── results.tex
```
where the child documents are added to "main.tex" via `\input{}`. 

"main.tex" looks like this: 

```
\begin{document}
Consider the lobster \citep{acheson1988lobster}
\section{Introduction}
\input{intro}
\section{Results}
\input{results}
\section{Conclusion}
\input{conclusion}
\bibliographystyle{chicago}
\bibliography{/Users/LawrenceDeGeest/Desktop/notebook/references}
\end{document}
```
and "intro.tex" looks like this:

```
%\bibliography{/Users/LawrenceDeGeest/Desktop/notebook/references}

Consider the lobster \citep{}
```

I add the commented-out references file at the top so I can use the wonderful auto-fill feature when typing `\citep{}`. However, *this* now produces the "no bib file found" message. Strange because it used to work just fine. 

Stranger yet is that copy-pasting a working file to a new, empty file in a new, empty directory causes it to break. Suppose I create a new directory called "mypaper2" and add a blank "main.tex" file, then copy the contents from `mypaper/main.tex` to `mypaper2/main.tex`. This should be fine, but instead it leads to an error. 

All these files compile accordingly with references included, whether I build from Sublime or from the command line.

**Comment by r-stein** [comment]
> and "intro.tex" looks like this:
>```latex
>%\bibliography{/Users/LawrenceDeGeest/Desktop/notebook/references}
>Consider the lobster \citep{}
>```
Writing `\bibliography` inside comments is not supported anymore.
The recommended way is to write `% !TEX root = main.tex` at the top of each included document. We may also provide a way to add some meta information in the comments.

The other seems to be a bug, if the same bibliography is used by two documents. We will have a look at this.

**Comment by lrdegeest** [comment]
Ok, thank you.

**Comment by KTGLeiden** [comment]
Just to let you know, I have the same issues with the latest Git version. Renaming the document to test.tex works great. Thanks for the fix and for looking into this.

**Comment by superjax** [comment]
I can confirm the same behavior on my machine. Renaming the document to test.tex fixed it.

**Comment by brainbug89** [comment]
I have the same problem. I getting the "No bib files found!" pop-up when trying to use `\cite{}`
Here is the output from the console on `zusammenfassung.tex`:
```
TEX root: 'D:\\privat\\xxx\\xxx\\xxx\\xxx\\xxx\\xxx\\version06\\contents\\zusammenfassung.tex'
Bib files found: 
[]
error: No bib files found!
```
The TEX root is wrong. Looks like the `"TEXroot": "main.tex",` setting at the .sublime-project doesn't work anymore. (btw `\cite{}` works fine at `main.tex`)

**Comment by lrdegeest** [comment]
Now bibliography commands like  `\citep{}` no longer brings up the list of references in the attached .bib file. This used to work fine days ago. Nothing changed on my end. This is getting quite frustrating. Is there any news with this?

**Comment by r-stein** [comment]
try to press C-l,C-backspace

**Comment by lrdegeest** [comment]
No effect. 

Also, I am getting the same "no bib files found!" pop-up on a beamer presentation. There is just one .tex file and the bibliography compiles fine.

**Comment by lrdegeest** [comment]
Oh, I forgot to mention that like before, re-naming the file to "test.tex" resolves the pop-up issue. Very bizarre.

**Comment by r-stein** [comment]
Not sure what's the issue. You can try to open the ST console and paste `import os; os.startfile(sublime.cache_path() + "/LaTeXTools")` this should open the cache folder. Just delete the local_cache folder and the files, which are called like `tradbib_XXXX` or `new_XXX`. Afterwards restart ST.

**Comment by lrdegeest** [comment]
But the bug persists even if I remove the package altogether and re-install. Other users are pointing out that renaming files to "test.tex" resolves the issue for them too.

**Comment by r-stein** [comment]
Reinstalling the package doesn't clear the cache. Not sure about the renaming it think it just clears the cache by using a different file path. We had the last update 12 days ago and did not change the cache in it, but we may just invalidate all old cache files in the next one.

**Comment by lrdegeest** [comment]
Renaming the file to "test.tex" really appears to be the only solution so far. Clearing cache has no effect. Do you have any insight at all as to why?

**Comment by r-stein** [comment]
Have you manually deleted those files? We don't have a special meaning for a file with the name "test.tex", but use the md5sum of the file path to identify the caches. I think it's more about renaming then about the file name.

**Comment by r-stein** [comment]
@lrdegeest I have relased a [prerelease](https://github.com/SublimeText/LaTeXTools/tree/v3.14.1-rc#prereleases), which slighty improves error handlings of the cache and adds a *LaTeXTools: Clear Cache* command, which superseeds the other clear cache commands and removes all cache files. It does not fix the underlaying issues, but you may try that if you still have the problems.

**Comment by ntvy95** [comment]
@r-stein I have met the same problem as lrdegeest. I installed the prerelease version but I could not find the Clear Cache command? Renaming the file to "test.tex" does the trick, and it turns out that it is the only name that works.

**Comment by larkinlheintzman** [comment]
**@r-stein** clearing the cache, via the Clear Cache command worked for me! Thanks for the fix.

**Comment by RAXYAO** [comment]
Clearing the cache and renaming the file both work for me.

**Comment by TaylanT** [comment]
Hey guys, same problem here.
My .bib file is in the same directory as my main.tex. Cache and document cache is cleared. The `\bibliography `command is called from a separate file (.sty - File) . 
`% !TEX root = ../my-thesis.tex` is included in the page header
renaming the file did not work for me. 
Unfortunately  Sublime / LatexTools  still gives me the popup "No bib files found!" .

**Comment by r-stein** [comment]
@TaylanT Calling `\bibliography` should behave the same no matter from where you call. We rely on a document-wide analysis, which is independent of the currently opened file. You may add `.sty` to your ["tex_file_exts"](https://github.com/SublimeText/LaTeXTools/blob/st3-3.14.1/LaTeXTools.sublime-settings#L67) though.

**Comment by Bankimchandrayadav** [comment]
Solution for Sublime Test 3
===========================

It generally appears when working in a structure of where main file is kept in one folder while child documents are kept in a subfolder

        -  directory
		-	main.tex
		-	(subdirectory)chapters
		-	(subdirectory)images

Individual chapters files are kept in chapters subdirectory.

Solution
========

-	Do not make individual chapter files rather write all chapter codes in main.tex 
-	Rename the file 'main.tex' to 'test.tex' (removes error for no bib file found)
-	Change bibliography style to apalike. later on change to whatever you wish to after 
	deleting the auxiliary files. (removes error for citation undefined)

The above three changes can guarantee the solution with 90% probability.

**Comment by babesyoung** [comment]
thx
% !TEX root = ***.tex work for me

**Comment by bbercovici** [comment]
I've been running into the same problem today: C-B was compiling a document along with its bibliography fine, but the autocomplete of \cite{ was sometimes failing with a "No bib file found" error message, depending upon the document. The bibliography was loaded in using its absolute path.

I could not understand why some of my documents were working fine, while others were not. After some fiddling, I can confirm that the master document's name matters. "poster.tex" does not let me use autocomplete within Sublime Text, but renaming it "postera.tex" does the trick. Modifying it back to "poster.tex" brings the error back. o_O

**Comment by r-stein** [comment]
If changing the document name helps try to press ctrl+shift+p and execute *LaTeXTools: Clear Cache*

**Comment by jorgeam11** [comment]
I've had same issue with everything working fine except for the \cite{} command giving me "No bib file found". Doing the _LaTeXTools: Clear Cache_ fixed the problem instantly thanks!

**Comment by lrdegeest** [comment]
Clearing the cache works, but why is it necessary to do so at the start of *every* project?

**Comment by r-stein** [comment]
I am not sure what the problem is and such problems are very hard to reproduce and hence to fix. I added that command to at least give the option to manually clear the cache if something went wrong and we need to rethink the cache structure, e.g. we could just keep the cache only in memory in some cases instead of additionally writing stuff to hard disk.

**Comment by ig0774** [comment]
I think #1192 *should* fix this, at least the version of this I've been able to reproduce. But moving to a strictly in-memory cache would also fix things as well.

**Comment by qiaojunfeng** [comment]
Yes, clear cache works for me too!

**Comment by kdzhang** [comment]
Same problem, and clearing cache works for me!

**Comment by gannebamm** [comment]
I can confirm that the following multi-file and directory structure works nicely on my Windows machine:
```
- main.tex
- literature.bib
-\chapters\
            | intro.tex
            | methods.tex
            | ....tex
```

If I start every chapters/....tex file with `% !TEX root = ..\main.tex`
Is there any documentation for LaTeXTools for handling multi-file projects? I have never seen this command before.

**Comment by r-stein** [comment]
You can find the documentation here https://latextools.readthedocs.io/en/latest/features/#multi-file-documents

You should use a normal slash in the path to the root file to be compatible with other OS and editors, I.e. `%!TEX root = ../main.tex`

Aside: you may also be interested that you can auto insert that command on file creation from the `\input`command [see](https://latextools.readthedocs.io/en/latest/keybindings/#jumping-to-included-files)

**Comment by kristtja** [comment]
This problem occurred to me today as well, but only after converting my citations from bibtex to biblatex. I've tried everything you guys have posted and nothing worked. The only way I managed to solve this was to revert everything back to bibtex... Are there any other things one might try to solve this problem?

**Comment by ig0774** [comment]
When you converted the file, how did you load the bibliography?

**Comment by ig0774** [comment]
E.g. `\addbibresource` must have the file name with the `.bib` extension, unlike with straight BibTeX.

**Comment by kristtja** [comment]
The bibliography is loaded in a custom .sty file with 

\RequirePackage[backend = biber, style = authoryear, maxnames = 2]{biblatex}
\addbibresource{bibliography.bib}
\renewcommand*{\nameyeardelim}{\addcomma\space}

and then added added in the main.tex file using \printbibliography.

For bibtex the entry in the .sty file is \RequirePackage{natbib} and then the main.tex file has
\bibliographystyle{myBibStyle.bst}
 \bibliography{bibliography}

What was really weird is that ST2 managed to find the .bib file the first 5-10 minutes after converting, and then the error suddenly appeared. The same happened when I tried to create new files in a different folder. I have also tried to change to ST3 but the error still happened.

My main.tex, .bib file, and .sty file are all in the same folder. The different sections lies in a subfolder. All documents in the subfolder has the line %!TEX root = ../main.tex as their first line.

**Comment by ig0774** [comment]
This has a fairly easy answer then: LaTeXTools doesn't currently support finding bibliographies defined in custom style files. Only those defined in the main document or files `\input`ed or `\import`ed will be found.

To get this to work, you'd need to add something like the following to your document preamble:

```latex
\newcommand{\nop}[1]{}
\nop{\addbibresource{bibliography.bib}}
```

**Comment by kristtja** [comment]
Added the commands to the preamble, but unfortunately nothing changed. 
I also tried to add the bibliography in the preamble using 

    \usepackage[backend = biber, style = authoryear, maxnames = 2]{biblatex}
    \bibliography{bibliography.bib}
    \renewcommand*{\nameyeardelim}{\addcomma\space}

but that didn't help either.

**Comment by ig0774** [comment]
Sorry, I forgot to add that you need to run the `LaTeXTools: Clear Cache` command after making the changes.

**Comment by kristtja** [comment]
Tried to clear the cache, but the error still occurs. Any other ideas?

**Comment by lrdegeest** [comment]
You may need to clear multiple caches

![screen shot 2018-02-07 at 9 00 11 am](https://user-images.githubusercontent.com/10857097/35920456-e70d9f90-0be5-11e8-8026-6a493b6e1eb3.png)

**Comment by kristtja** [comment]
Still not working :( I might just have to convert to biblatex when I'm done with my thesis.

**Comment by cipri-tom** [comment]
**TL;DR:** check you correctly specify the TEX ROOT (`main.tex`) path in your sub-files !

I ran into this error today and clearing the cache didn't solve it. But then I checked the console ``ctrl+` `` and I saw my `main.tex` wasn't found. Specifically, it was searching for `..../thesis/Chapters/main.tex` whereas my `main.tex` was in the root `..../thesis`. Easy to spot, therefore: At the beginning of the chapter file I had `%!TEX root = main.tex`, whereas I should've had `%!TEX root = ../main.tex`

<sup><sup>Markdown is hard: [``ctrl+` `` == ``` ``ctrl+` `` ```](https://meta.stackexchange.com/a/70679/221120)</sup></sup>

**Comment by justinesarey** [comment]
Getting "no bib file found" error. Clearing all caches does not appear to remedy the error. Console returns:

TEX root: 'C:\\Dropbox\\jesarey_documents\\Gender Corruption Instrument\\2018-3-5_gender-corruption-and-causality-presentation.tex'
Bib files found: 
[]
error: No bib files found!

...so TEX root is correctly identified (that is, in fact, the location of the master .tex file). Weirdly, this was working for a while and then suddenly stopped working. Now I can't get it working again.

**Comment by piggy00wiggy** [comment]
I have two projects which share the same bib file via absolute path, and of course the \bibliography{} in both files are the same.

however, one can pop up the bib list successfully, another cannot

**Comment by Whadup** [comment]
> Sure, but one last point. The problem is also sensitive to the _name_ of a file. "test.tex" works; "experiment.tex" or "asasdfl.tex" (random letters) do not.

I've had the same behavior, main.tex seems to work reliably, anything else might fail.

**Comment by pabloarcadio** [comment]
The same message "No bib files found" when writing \cite{} in the chapter.tex. I understand after reading the comments that this is due to the use of .tex as chapters (inside the main.tex  comand \cite{} works perfectly). Then I did what user @cipri-tom commented about placing this line in every chapter: 
%!TEX root = ../main.tex
but still is not working at goes on the message of "Non bib file found". Any ideas?

**Comment by r-stein** [comment]
@pabloarcadio what is the folder structure of your project? Are the chapters in a subfolder of the master document?

```
project/
├── chapters/
│   ├── chapter.tex
│   └── other-chapter.tex
└── main.tex
```

**Comment by pabloarcadio** [comment]
Yes @r-stein , it is almost like that: there are folders inside chapters/ 
project/
         chapters/
                      chapter01/
                                      chapter01.tex
         main,tex

**Comment by r-stein** [comment]
In that folder structure you need to change it to 

```
%!TEX root = ../../main.tex
```

I.e. the relative path to the main file.

**Comment by pabloarcadio** [comment]
Sorry, buy I don't understand what you mean. I wrote at the beggining (just before the \begin{document} of the main.tex this:
%!TEX root = D:/Doctorado/Tesis/Tesis Latex/memoirthesis.tex
and inside the chapter01.tex the same:
%!TEX root = D:/Doctorado/Tesis/Tesis Latex/memoirthesis.tex
%
\let\textcircled=\pgftextcircled
\chapter{Prólogo}
\label{chap:intro}

**It is that approach correct?** 
Thanks

**Comment by r-stein** [comment]
I am not sure whether your `main.tex` file differs from `memoirthesis.tex`, The tex root should be the file you always compile and it is in fact used when you compile via `C-b`.
You can check the tex root view opening the ST console ``C-` `` and then writing:

```
from LaTeXTools.latextools_utils import tex_directives
tex_directives.get_tex_root(view)
```

I don't know of there are problems with absolute paths or spaces in the path, but that setting looks correct to me.
Note that you don't need to set the main file inside your main file.

**Comment by pabloarcadio** [comment]
Thanks @r-stein. I have solved the problem and now I can see the bibliography list in every chapter. The solution is to place as the root not only the main file:
%!TEX root = ../../main.tex
but as well the biblio file:
%!TEX root = ../../biblio.tex

Don't ask me why but it works!

**Comment by r-stein** [comment]
@pabloarcadio If you use the same tex directive twice one will win, it will just use `biblio.tex` as the tex root, which is okay for the `\cite` command, but will interfere with other features (quick build, command completions, ...). If that's okay for you can do it like this.

**Comment by mmauii** [comment]
> # Solution for Sublime Text 3
> It generally appears when working in a structure where main file is kept in one folder while child documents are kept in a subfolder
> 
> ```
>     -  directory
> 	-	main.tex
> 	-	(subdirectory)chapters
> 	-	(subdirectory)images
> ```
> Individual chapters files are kept in chapters subdirectory.
> 
> # Solution
> * Do not make individual chapter files rather write all chapter codes in main.tex
> * Rename the file 'main.tex' to 'test.tex' (removes error for no bib file found)
> * Change bibliography style to apalike. later on change to whatever you wish to after
>   deleting the auxiliary files. (removes error for citation undefined)
> 
> The above three changes can guarantee the solution with 90% probability.

I wanted to keep making individual chapter files to keep the main.tex file clean. The issue remained tho; while trying the \cite{} command inside an individual chapter file, the "no bib files found" pop-up showed up.

What solved it for me, beside naming main.tex into test.tex, was separating the whole preamble into a header.tex file which is then included in the test.tex file via the \input{} command. The \addbibresource{} is inside the header.tex file.

**Comment by lrdegeest** [comment]
Somehow this is *still* a problem. Just today I fired up a brand new document and got the dreaded "No bib files found!" 

Here is an example:

```
\documentclass{article}
\usepackage{natbib}
\begin{document}
Consider the lobster \citep{acheson1988lobster}
\bibliographystyle{chicago}
\bibliography{references}
\end{document}
```

As usual, clearing the caches does nothing. 

Will this bug ever be fixed?

**Comment by ig0774** [comment]
@lrdegeest  So, just to be clear, the issue here isn't one bug, but a series of different configurations or setups that sometimes don't play well with LaTeXTools. The problem is generating a reproducible example of an issue so we can fix it.

For example, on the latest pre-release, with a file named `references.bib` in the same directory as my saved `tex` document with the contents you gave, I don't get the "No bib files found!" message. Could you please post the contents of your ST console when that error occurs? We might be better able to track down what is happening from there.

**Comment by kristtja** [comment]
Hi again,
I'm trying to give biblatex another chance, but yet again I'm getting the "No bib files found" error. What's extra interesting is that I'm now on a Windows machine and not a Mac. 

The \addbibresource is in the preamble of the main-file, the subsections all have %!TEX root = ../main.tex in the top, and of course the bib-file is in the same folder as the main-file. I've also tried to clear the cache multple times.

I've tried ignoring the error and writing down the reference key and get the error that the citation is undefined.

When the "No bib files found" error shows uo, the console says:
TEX root: 'C:\\Users\\cvkt5\\OneDrive - Loughborough University\\PhDThesis\\main.tex'
Bib files found: 
()
error: No bib files found!

So at least it's pointing to the correct path to the main-file

**Comment by samuelbl** [comment]
add "%! TEX root = ../main.tex" on all files that are not in the home directory worked for me

**Comment by 9yifanchen9** [comment]
For a single.tex file, changing the file name to not contain any spaces worked for me

**Comment by KSokol79** [comment]
> add "%! TEX root = ../main.tex" on all files that are not in the home directory worked for me

Does this approach allow you to compile the subfiles? I am under the impression putting %! TEX root at the top of the subfile overrides the setting. Has the workaround been developed?

**Comment by thistlillo** [comment]
> Can you press C-l, C-d, C-c and then check whether it still doesn't work?

I had the same issue (no bib file found, but it was well present) and this solved the issue in my case.

**Comment by ultravioletcatastrophe** [comment]
This seems to still be an issue, and the only thing that seems to work is manually generating a .bbl file via bibtex ______

**Comment by connbrack** [comment]
Just solved my issue, likely not a universal fix, but in case it helps someone:

I import my packages from an .sty file, where I had my \addbibresource{bibliography.bib}

When i moved my \addbibresource{bibliography.bib} to my main .tex file my issue was solved. BUT only after i closed and reopened sublime (I would have figured this out much earlier, but when I tried this before i didn't restart sublime).

I also removed all my build files before restarting. Not sure if this was required.

**Comment by githubedu** [comment]
> When i moved my \addbibresource{bibliography.bib} to my main .tex file my issue was solved. BUT only after i closed and reopened sublime (I would have figured this out much earlier, but when I tried this before i didn't restart sublime).

Yes, after trying many recommendations that has been mentioned before. Closing and opening sublime solve the problem. My advice is: Try something, then close and open sublime.
